src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -301,6 +302,43 @@ public void setServingBorrower(BorrowerId borrowerId) {
         this.servingBorrower = Optional.of(borrowerRecords.getBorrowerFromId(borrowerId));
     }
 
+    @Override
+    public void setServingBorrower(Borrower borrower) {
+        this.servingBorrower = Optional.of(borrower);
+    }
+
+    @Override
+    public void servingBorrowerNewLoan(Loan newLoan) {
+        if (!isServeMode()) {
+            throw new NotInServeModeException();
+        }
+
+        Borrower serving = servingBorrower.get();
+        Borrower loanAddedBorrower = new Borrower(serving.getName(), serving.getPhone(), serving.getEmail(),
+                serving.getBorrowerId(), serving.getAddedCurrentLoanList(newLoan), serving.getReturnedLoanList());
+        borrowerRecords.setBorrower(serving, loanAddedBorrower);
+
+        setServingBorrower(loanAddedBorrower);
+    }
+
+    @Override
+    public void servingBorrowerReturnLoan(Loan returnedLoan) {
+        if (!isServeMode()) {
+            throw new NotInServeModeException();
+        }
+
+        Borrower serving = servingBorrower.get();
```

Comment:
Would you consider checking `servingBorrower.isPresent()` here before getting the value from the optional just to be safe

Reply:
This method will only be called when the model is in serve mode, i.e., when the servingBorrower is present. The check for this is done in logic but I already added another safeguard check to throw NotInServeModeException. Thus, I think there's not need to check this again for the third time.